### PR TITLE
Fix save_to for Mock

### PR DESCRIPTION
**Pull request by ivank**
Great project, just a small fix for an issue I ran into while using your library - one cannot use "save_to" and write proper tests for it at the moment. 

This allows using “save_to” in mocked requests. Now you can write tests for code that uses “save_to” easily.
Test included.

### CODE REVIEW

File: src/Subscriber/Mock.php [hunk 1]
```diff
@@ -60,6 +60,12 @@ public function onBefore(BeforeEvent $event)
             }
         }
 
+        $save_to = $event->getRequest()->getConfig()->get('save_to');
+
+        if ($save_to) {
+            file_put_contents($save_to, $item->getBody());
```
Comment (mtdowling) [inline src/Subscriber/Mock.php]: save_to can be of various types: http://docs.guzzlephp.org/en/latest/clients.html#save-to. I think this method would need to account for that.

Comment (ivank) [inline src/Subscriber/Mock.php]: Yeah, sorry about that, totally forgot. here's the updated PR with all the 3 use cases

File: src/Subscriber/Mock.php [hunk 2]
```diff
@@ -60,6 +61,20 @@ public function onBefore(BeforeEvent $event)
             }
         }
 
+        $save_to = $event->getRequest()->getConfig()->get('save_to');
```
Comment (GrahamCampbell) [inline src/Subscriber/Mock.php]: Shouldn't the variable be called `$saveTo`?

Comment (ivank) [inline src/Subscriber/Mock.php]: I think scrutinizer could be configured to spot (and create patches for) these kinds of style violations automatically. Anyway fixed.

File: src/Subscriber/Mock.php [hunk 3]
```diff
@@ -60,6 +61,20 @@ public function onBefore(BeforeEvent $event)
...
+
+        if (null !== $saveTo) {
+        	$body = $item->getBody();
+
+        	if (is_resource($saveTo)) {
+        		fwrite($saveTo, $body);
+        	} elseif (is_string($saveTo)) {
+	            file_put_contents($saveTo, $body);
+        	} elseif ($saveTo instanceof StreamInterface) {
+        		$saveTo->write($body);
```
Comment (mtdowling) [inline src/Subscriber/Mock.php]: Can you fix the indentation here to match the rest of the library (4 spaces)?

**Comment by mtdowling** [comment]
Looks good! Just need a minor tweak to fix the spacing, but other than that, LGTM.

**Pull request merged**